restapi/opts.go

```diff
@@ -70,3 +66,51 @@ func Retry(n int) Option {
 		return client
 	}
 }
+
+// UseConfigFile setup rest client from toml file
+func UseConfigFile(path *string) Option {
+	return func(client *tClient) *tClient {
+		type config struct {
+			BaseURL     string       `toml:"base_url"`
+			Certificate *Certificate `toml:"api_ca_crt"`
+		}
+		var file struct {
+			API config
+		}
+
+		if path == nil {
+			return client
+		}
+
+		f, err := os.Open(*path)
+		if err != nil {
+			panic(err)
+		}
+		defer f.Close()
+
+		data, err := ioutil.ReadAll(f)
+		if err != nil {
+			panic(err)
+		}
+
+		err = toml.Unmarshal(data, &file)
+		if err != nil {
+			return client
+		}
+
+		client = BaseURL(file.API.BaseURL)(client)
+		client = X509(file.API.Certificate.X509)(client)
+		return client
+	}
+}
+
+// UseEnvironment setups rest client using environment variables
+func UseEnvironment() Option {
+	return func(client *tClient) *tClient {
```

Comment:
Should we allow trust anchor certificate to be communicated via environment for symmetry (either as 64, or as a file path), as now it needs either to come from the configuration file or from the system settings.?

Reply:
I though about this. We have a limit on env variable size
> The theoretical maximum length of an environment variable is around 32,760 characters. However, you are unlikely to attain that theoretical maximum in practice. All environment variables must live together in a single environment block, which itself has a limit of 32767 characters.

I'd prefer to use external file instead of inline variable value. If file is used than it does not differ from config file and we have support for this already. I'd keep Env vars only for the case when CA is trusted via OS trust anchor, which is true for all cases expect self-signed certificate.

Reply:
I would not do anything specifically for this case.

Reply:
Fair enough.